inference-engine/src/mkldnn_plugin/nodes/mkldnn_concat_node.cpp

```diff
@@ -27,291 +27,307 @@ using namespace mkldnn;
 using namespace MKLDNNPlugin;
 using namespace InferenceEngine;
 
+
+bool MKLDNNConcatNode::isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        errorMessage = std::string("CPU plugin doesn't support nGraph operation ") + op->get_type_name() + " with name '" + op->get_friendly_name() + "' ";
+        auto concatOp = ngraph::as_type_ptr<const ngraph::op::v0::Concat>(op);
+        if (!concatOp) {
+            return false;
+        }
+        if (concatOp->get_input_size() == 0) {
+            errorMessage += "and inputs num " + std::to_string(concatOp->get_input_size()) + ". Two inputs expected.";
+            return false;
+        }
+        if (concatOp->get_output_size() != 1) {
+            errorMessage += "and outputs num " + std::to_string(concatOp->get_input_size()) + ". One output expected.";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
 MKLDNNConcatNode::MKLDNNConcatNode(const std::shared_ptr<ngraph::Node>& op, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache)
-        : MKLDNNNode(op, eng, cache) {}
+        : MKLDNNNode(op, eng, cache) {
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        THROW_IE_EXCEPTION_WITH_STATUS(NOT_IMPLEMENTED) << errorMessage;
+    }
+
+    auto concatOp = ngraph::as_type_ptr<ngraph::op::v0::Concat>(op);
+    axis_ = concatOp->get_axis();
```

Comment:
The specification says that negative axis is supported. Do we handle such case?

Reply:
Fixed